test_runner/regress/test_truncate.py

```diff
@@ -0,0 +1,47 @@
+import time
+
+from fixtures.neon_fixtures import NeonEnvBuilder
+
+
+#
+# Benchmark searching the layer map, when there are a lot of small layer files.
+#
+def test_truncate(neon_env_builder: NeonEnvBuilder, zenbenchmark):
+
+    env = neon_env_builder.init_start()
+    n_records = 10000
+    n_iter = 10
+
+    # We want to have a lot of lot of layer files to exercise the layer map. Make
+    # gc_horizon and checkpoint_distance very small, so that we get a lot of small layer files.
+    tenant, _ = env.neon_cli.create_tenant(
+        conf={
+            "gc_period": "100 m",
+            "gc_horizon": "1048576",
+            "checkpoint_distance": "1000000",
+            "compaction_period": "1 s",
+            "compaction_threshold": "3",
+            "image_creation_threshold": "1",
+            "compaction_target_size": "1000000",
+        }
+    )
+
+    env.neon_cli.create_timeline("test_truncate", tenant_id=tenant)
+    pg = env.postgres.create_start("test_truncate", tenant_id=tenant)
+    cur = pg.connect().cursor()
+    cur.execute("set statement_timeout=0")
+    cur.execute("create table t1(x integer)")
+    cur.execute(f"insert into t1 values (generate_series(1,{n_records}))")
+    cur.execute("vacuum t1")
+    for i in range(n_iter):
+        cur.execute(f"delete from t1 where x>{n_records//2}")
+        cur.execute("vacuum t1")
```

Comment:
It is difficult to insert some strict checks here because all operations are non-deterministic.
Order of storing records in relation is not specified, so there is absolutely no warranty that after insertion some number of records and them deleting some some part of them relation size will really decrease.

Reply:
Sorry, but TRUNCATE actually doesn't log truncation records in WAL.
It just removes relation files.

Reply:
> So, we don't really test truncation codepath, but only the part that zeroes out the page, right?

Before this PR we truncate FSM/VM fork to zero in this case. And it cause "key not found" error.
So the main goal of this test was to demonstrate that this error is not happening any more.
Yes, used table size is too small to cause truncation of FSM fork. But if we increase it, then it really happen:
```
postgres=# insert into t1 values (generate_series(10000/2+1,1000000));
INSERT 0 995000
postgres=# select pg_relation_size('t1', 'fsm')/8192;
 ?column? 
----------
        4
(1 row)

postgres=# delete from t1 where x>10000/2;
DELETE 1090000
postgres=# vacuum t1;
VACUUM
postgres=# select pg_relation_size('t1', 'fsm')/8192;
 ?column? 
----------
        3
(1 row)
```